ios/PhotoLibrary/RNPermissionHandlerPhotoLibrary.m

```diff
@@ -56,9 +56,17 @@ - (void)openLimitedPhotoLibraryPickerWithResolver:(RCTPromiseResolveBlock _Nonnu
     if ([PHPhotoLibrary authorizationStatusForAccessLevel:PHAccessLevelReadWrite] != PHAuthorizationStatusLimited) {
       return reject(@"cannot_open_limited_picker", @"Photo library permission isn't limited", nil);
     }
+    
+    UIViewController* topViewController = [[UIApplication sharedApplication].keyWindow rootViewController];
+    UIViewController* presentedViewController = topViewController.presentedViewController;
+
+    while (presentedViewController = topViewController.presentedViewController) {
+      topViewController = presentedViewController;
+      presentedViewController = topViewController.presentedViewController;
+    }
 
     UIViewController* rootViewController = [[UIApplication sharedApplication].keyWindow rootViewController];
-    [[PHPhotoLibrary sharedPhotoLibrary] presentLimitedLibraryPickerFromViewController:rootViewController];
+    [[PHPhotoLibrary sharedPhotoLibrary] presentLimitedLibraryPickerFromViewController:topViewController];
```

Comment:
Add the import:

```objc
#import <React/RCTUtils.h>
```

Replace all of this with:

```objc
UIViewController *topViewController = RCTPresentedViewController();
[[PHPhotoLibrary sharedPhotoLibrary] presentLimitedLibraryPickerFromViewController:topViewController];
```

Reply:
I agree, that would definitely be the better way to do it. Tested and it works!